TeamCharles/workforce-management
# 14-employee-creation

## Status
**READY**

## Documentation

[X] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[ ] This PR does **NOT** require documentation.

## Description
Provide a form to the Employee/Add view that allows a user to create a new Employee. The Employee must be assigned a computer and a department, and may optionally be added to one or more training programs. On successful creation, you should be routed to a detail view for that newly created user.

## Resolves Issue Number
Resolves #14 and #19 

## Deploy Notes

```sh
git pull --prune
git checkout 14-employee-creation
dotnet run
```

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

1. Navigate to localhost/Employee/Add
1. Fill out the form
1. The "Add Employee" button should be disabled until all of the required fields are completed
1. You should only be able to select from unassigned Computers
1. Validation should work
1. After submitting the form, a new Employee record should be created in your database
1. If you selected any training programs, new Attendee records should be created

## Impacted Files in Application
List general components of the application that this PR will affect:

* Views/Employee/Add
* Controllers/EmployeeController
* ViewModels/Employee/EmployeeForm
* site.js
